testing/terraform/ec2/main.tf

```diff
@@ -27,8 +27,16 @@ locals {
 }
 
 data "aws_ami" "ami" {
-  executable_users = ["self"]
+  owners = ["amazon"]
   most_recent      = true
+  filter {
+    name   = "name"
+    values = ["al20*-ami-minimal-*-x86_64"]
```

Comment:
NB comment: Do we need to specify `x86_64` in the name even though we have additional filter on the architecture down below?

Reply:
Also, how have we verified that the AMI identified here will always be usable?

Reply:
Prashant: I believe it's redundant, is this something we prefer to have removed?
Mahad: There is a depreciation date for each ami, usually around ~3 month. It looks like a new version of this minimal AMI is released every ~2 weeks (Period is kind of infrequent) and I can see it going back to Oct 2023. (Further back was probably depreciated so I can't see). I think this AMI has been being maintained for a while and we can expect newer versions to keep getting released.

Reply:
The AMI name is defined by convention. I find one public doc talked about it [here](https://docs.aws.amazon.com/linux/al2023/ug/ec2.html#launch-from-ec2-console)

>al2023-[ami || ami-minimal]-2023.0.[release build date].[build number]-kernel-[version number]-[arm64 || x86_64]

I think we can mostly assume the convention will be kept. Though no perfect solution here.